source/Ship.cpp

```diff
@@ -1625,13 +1658,35 @@ void Ship::DoGeneration()
 		if(shieldsAvailable)
 			heat += (shieldsAvailable - shieldsRemaining) * shieldsHeat;
 	}
+	
 	// Handle ionization effects, etc.
+	// TODO: Mothership gives status resistance to carried ships?
 	if(ionization)
-		ionization = max(0., .99 * ionization - attributes.Get("ion resistance"));
+	{
+		const double ionResistance = attributes.Get("ion resistance");
```

Comment:
Making these const here, but not in the method called with them, is meaningless. With the new reduced scope, you can discard the const qualifier (or update the called method's signature).